File: frontend/src/app/models/orderdetail/OrderDetail.ts
import { BaseEntity } from '../base/BaseEntity';
import { Product } from '../product/Product';
import { Order } from '../order/Order';

export class OrderDetail extends BaseEntity {
  quantity: number;
  totalPricePerProduct: number;
  product: Product;
  order: Order;

  constructor(
    quantity: number,
    totalPricePerProduct: number,
    product: Product,
    order: Order
  ) {
    super();
    this.quantity = quantity;
    this.totalPricePerProduct = totalPricePerProduct;
    this.product = product;
    this.order = order;
  }
}

File: backend/src/main/java/com/springbootjwtpostgres/backend/orderdetail/OrderDetail.java
package com.springbootjwtpostgres.backend.orderdetail;

import com.springbootjwtpostgres.backend.basemodels.BaseEntity;
import com.springbootjwtpostgres.backend.order.Order;
import com.springbootjwtpostgres.backend.product.Product;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;


@Entity
@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
public class OrderDetail extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int quantity;
    private double price;
    private double totalPricePerProduct;
    private Date createdAt;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "product_id")
    private Product product;
    @Column(name = "product_id", insertable = false, updatable = false)
    private Long productId;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "order_id")
    private Order order;
    @Column(name = "order_id", insertable = false, updatable = false)
    private Long orderId;
}
